### PR TITLE
Add default updated_at column

### PR DESCRIPTION
**Pull request by coorasse**
Fixes issue #954 .

### CODE REVIEW

**Comment by coorasse** [comment]
Hello @joshuaclayton . Do I have to fix Houd comments?

File: lib/factory_girl/strategy/stub.rb [hunk 1]
```diff
@@ -75,6 +75,28 @@ def created_at
...
         end
+
+        responds_to_updated_at = result_instance.respond_to?(:updated_at)
+        updated_at_missing_default = responds_to_updated_at && !result_instance.updated_at
+        result_instance_missing_updated_at = !responds_to_updated_at
+
+        if updated_at_missing_default || result_instance_missing_updated_at
+          result_instance.instance_eval do
+            def updated_at
+              @updated_at ||= Time.now.in_time_zone
```
Comment (joshuaclayton) [inline lib/factory_girl/strategy/stub.rb]: Thoughts on using `Time.current`, which is Rails' standard that takes timezones into account?

File: spec/acceptance/build_stubbed_spec.rb [hunk 2]
```diff
@@ -150,7 +150,53 @@
   end
 
   it "doesn't allow setting created_at on an object that doesn't define it" do
-    expect { build_stubbed(:thing_without_timestamp, :created_at => Time.now) }.to raise_error(NoMethodError, /created_at=/)
+    expect { build_stubbed(:thing_without_timestamp, :created_at => Time.now) }
```
Comment (joshuaclayton) [inline spec/acceptance/build_stubbed_spec.rb]: Same comment here re: `Time.now` -> `Time.current`

File: spec/acceptance/build_stubbed_spec.rb [hunk 3]
```diff
@@ -150,7 +150,53 @@
...
+    FactoryGirl.define do
+      factory :thing_with_timestamp
+      factory :thing_without_timestamp
+    end
+
+    Timecop.freeze 2012, 1, 1
+  end
+
+  it "defaults updated_at for objects with updated_at" do
+    expect(build_stubbed(:thing_with_timestamp).updated_at).to eq Time.now
```
Comment (joshuaclayton) [inline spec/acceptance/build_stubbed_spec.rb]: `Time.now` -> `Time.current`?

File: lib/factory_girl/strategy/stub.rb [hunk 4]
```diff
@@ -75,6 +75,28 @@ def created_at
...
+            end
+          end
+        end
+      end
+
+      def add_missing_attribute(result_instance, column_name)
```
Comment (joshuaclayton) [inline lib/factory_girl/strategy/stub.rb]: Looks like this isn't used anywhere - should this get removed?

File: lib/factory_girl/strategy/stub.rb [hunk 5]
```diff
@@ -75,6 +75,28 @@ def created_at
             end
           end
         end
+
+        responds_to_updated_at = result_instance.respond_to?(:updated_at)
+        updated_at_missing_default = responds_to_updated_at && !result_instance.updated_at
+        result_instance_missing_updated_at = !responds_to_updated_at
+
+        if updated_at_missing_default || result_instance_missing_updated_at
```
Comment (joshuaclayton) [inline lib/factory_girl/strategy/stub.rb]: This brings up an interesting situation - why would we add `updated_at` to an object that doesn't already have that method defined?

File: spec/acceptance/build_stubbed_spec.rb [hunk 6]
```diff
@@ -150,7 +150,53 @@
...
+  end
+
+  it "defaults updated_at for objects with updated_at to the correct time with zone" do
```
Comment (joshuaclayton) [inline spec/acceptance/build_stubbed_spec.rb]: I think by switching to `Time.current` everywhere, we can remove this

File: lib/factory_girl/strategy/stub.rb (hunk 5 above)
Comment (coorasse) [inline lib/factory_girl/strategy/stub.rb]: Nice question. I obviously kept the behaviour consistent with created_at behaviour. Let me know what would you prefer. We would have to make them consistent anyway.

File: lib/factory_girl/strategy/stub.rb (hunk 4 above)
Comment (coorasse) [inline lib/factory_girl/strategy/stub.rb]: sorry. my bad. forgot here.

**Comment by coorasse** [comment]
I would prefer to replace `Time.current` for created_at in a separate PR.

**Comment by coorasse** [comment]
Back to you @joshuaclayton . There are a couple of points which I found difficult in this PR and I'd like to feedback you:
* Hound only runs in the PR: I cannot simulate it locally (or I don't know how to do that, so I need to commit and push to see if Hound will accept it)
* Line length set to 80 is very short IMHO (I would suggest to set it to 120 now that the project doesn't respect it anyway)
* We need an editor config file so I have the preferences set correctly (for example for the line length)

I'd glad to keep helping. I'll make another PR to use `Time.current` also in `created_at`.
Thanks

**Comment by coorasse** [comment]
Hei there @joshuaclayton , any update on this one? I was waiting for a feedback. Thanks!

**Comment by joshuaclayton** [comment]
@coorasse sorry for the late reply. Noted on the line length, although I think we should stick with 80 as a general rule of thumb, and opt to "violate" if we want. In that regard, I think this should be set. Mind squashing into one commit in preparation for merge?

**Comment by coorasse** [comment]
squashed and ready

**Comment by joshuaclayton** [comment]
@coorasse thank you for putting this together; excited to have this merged!

**Pull request merged**